base/sparse/sparsematrix.jl

```diff
@@ -964,25 +970,32 @@ sparse array will not contain any nonzero values. No storage will be allocated
 for nonzero values during construction. The type defaults to `Float64` if not
 specified.
 """
-spzeros(m::Integer, n::Integer) = spzeros(Float64, m, n)
-spzeros(Tv::Type, m::Integer, n::Integer) = spzeros(Tv, Int, m, n)
-function spzeros(Tv::Type, Ti::Type, m::Integer, n::Integer)
-    ((m < 0) || (n < 0)) && throw(ArgumentError("invalid Array dimensions"))
-    SparseMatrixCSC(m, n, ones(Ti, n+1), Array{Ti}(0), Array{Tv}(0))
-end
+spzeros(m::Integer, n::Integer) = zeros(SparseMatrixCSC, m, n)
+spzeros(Tv::Type, m::Integer, n::Integer) = zeros(SparseMatrixCSC{Tv}, m, n)
+spzeros(Tv::Type, Ti::Type, m::Integer, n::Integer) = zeros(SparseMatrixCSC{Tv,Ti}, m, n)
 
+function eye{Tv,Ti}(::Type{SparseMatrixCSC{Tv,Ti}}, m::Integer, n::Integer)
+    check_array_size(m, n)
+    x = min(m,n)
+    rowval = collect(Ti, 1:x)
+    colptr = Ti[rowval; fill(Ti(x+1), n+1-x)]
+    nzval  = ones(Tv, x)
+    return SparseMatrixCSC(m, n, colptr, rowval, nzval)
+end
+eye{Tv}(::Type{SparseMatrixCSC{Tv}}, m::Integer, n::Integer) = eye(SparseMatrixCSC{Tv,Int}, m, n)
+eye(::Type{SparseMatrixCSC}, m::Integer, n::Integer) = eye(SparseMatrixCSC{Float64,Int}, m, n)
 
-speye(n::Integer) = speye(Float64, n)
-speye(T::Type, n::Integer) = speye(T, n, n)
-speye(m::Integer, n::Integer) = speye(Float64, m, n)
+speye(n::Integer) = eye(SparseMatrixCSC, n)
+speye(T::Type, n::Integer) = eye(SparseMatrixCSC{T}, n, n)
+speye(m::Integer, n::Integer) = eye(SparseMatrixCSC, m, n)
 
 """
     speye(S)
 
 Create a sparse identity matrix with the same structure as that of  `S`.
```

Comment:
Right. Same nonzero structure wouldn't make much sense anyway. Good opportunity to correct that.